src/Common/src/TypeSystem/Ecma/EcmaModule.cs

```diff
@@ -352,11 +353,24 @@ private Object ResolveMemberReference(MemberReferenceHandle handle)
                 }
                 else
                 {
-                    MethodDesc method = ((TypeDesc)parent).GetMethod(name, parser.ParseMethodSignature());
-                    if (method != null)
-                        return method;
+                    MethodSignature sig = parser.ParseMethodSignature();
+                    TypeDesc typeDescToInspect = parentTypeDesc;
 
-                    // TODO: Lookup in parent
+                    // Try to resolve the name and signature in the current type, or any of the base types.
+                    do
+                    {
+                        MethodDesc method = typeDescToInspect.GetMethod(name, sig);
+                        if (method != null)
+                        {
+                            // If this resolved to one of the base types, make sure it's not a constructor.
+                            // Constructors are not inherited.
+                            if (typeDescToInspect != parentTypeDesc && (method.IsConstructor || method.IsStaticConstructor))
```

Comment:
The other type system implementations filter out instance constructors only for this case.